akka-docs/rst/additional/osgi.rst

```diff
@@ -1,12 +1,104 @@
+
+3. STARTING: A bundle that is started can be used by other bundles. For an otherwise complete application closure of
+   resolved bundles, the implication here is they must be started in the order directed by a depth-first search for all to
+   be started. When a bundle is starting, any exposed lifecycle interfaces in the bundle are called, giving the bundle
+   the opportunity to start its own service endpoints and threads.
+
+4. ACTIVE: Once a bundle's lifecycle interfaces return without error, a bundle is marked as active.
+
+5. STOPPING: A bundle that is stopping is in the process of calling the bundle's stop lifecycle and transitions back to
+   the RESOLVED state when complete. Any long running services or threads that were created while STARTING should be shut
+   down when the bundle's stop lifecycle is called.
+
+6. UNINSTALLED: A bundle can only transition to this state from the INSTALLED state, meaning it cannot be uninstalled
+   before it is stopped.
+
+Note the dependency in this FSM on lifecycle interfaces. While there is no requirement that a bundle publishes these
+interfaces or accepts such callbacks, the lifecycle interfaces provide the semantics of a ``main()`` method and allow
+the bundle to start and stop long-running services such as REST web services, ActorSystems, Clusters, etc.
+
+Secondly, note when considering requirements and capabilities, it's a common misconception to equate these with repository
+dependencies as might be found in Maven or Ivy. While they provide similar practical functionality, OSGi has several
+parallel type of dependency (such as Blueprint Services) that cannot be easily mapped to repository capabilities. In fact,
+the core specification leaves these facilities up to the container in use. In turn, some containers have tooling to generate
+application load descriptors from repository metadata.
+
+Notable Behavior Changes
+------------------------
+
+Combined with understanding the bundle lifecycle, the OSGi developer must pay attention to sometimes unexpected behaviors
+that are introduced. These are generally within the JVM specification, but are unexpected and can lead to frustration.
+
+* Bundles should not export overlapping package spaces. It is not uncommon for legacy JVM frameworks to expect plugins
+  in an application composed of multiple JARs to reside under a single package name. For example, a frontend application
+  might scan all classes from ``com.example.plugins`` for specific service implementations with that package existing in
+  several contributed JARs.
+
+  While it is possible to support overlapping packages with complex manifest headers, it's much better to use non-overlapping
+  package spaces and facilities such as `Akka Cluster`_
+  for service discovery. Stylistically, many organizations opt to use the root package path as the name of the bundle
+  distribution file.
+
+.. _Akka Cluster: @github@/akka-docs/rst/scala/code/docs/akka/current/common/cluster.html
```

Comment:
Is this link valid? We don't have such file in the repo. If you mean to link to docs then it should link via :refs: instead, shouldn't it?